service/src/lib.rs

```diff
@@ -129,15 +129,9 @@ impl<Config: Clone + Send + 'static, S: Service<Config>, F: Fn()> ConnectionMana
                 Box::new(constructor),
             );
             if let Err(outer) = service.start().await {
-                match outer {
-                    Error::BitcoinError(ref inner) if inner.is_transport_error() || inner.is_json_decode_error() => {}
-                    Error::RuntimeError(RuntimeError::ChannelClosed) => (),
-                    Error::RuntimeError(ref inner) if inner.is_rpc_error() => (),
-                    other => return Err(other),
```

Comment:
Do you remember why we filtered this?

Reply:
I'm not following. It looks like we didn't use to restart for non-disconnect errors?

Reply:
No errors are / were propagated from `run_and_monitor_tasks` [here](https://github.com/interlay/interbtc-clients/blob/5537c1cb874855d95d2a2a8212a3d45a55bc7cbd/vault/src/system.rs#L746) so the only thing this filters is disconnect errors from anything before that LoC.